control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2961,3 +2975,39 @@ func (r *HostedControlPlaneReconciler) removeCloudResources(ctx context.Context,
 	}
 	return false, nil
 }
+
+func (r *HostedControlPlaneReconciler) reconcileCSISnapshotControllerOperator(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImage *releaseinfo.ReleaseImage, createOrUpdate upsert.CreateOrUpdateFN) error {
+	params := snapshotcontroller.NewParams(hcp, releaseImage.Version(), releaseImage.ComponentImages(), r.SetDefaultSecurityContext)
+
+	deployment := manifests.CSISnapshotControllerOperatorDeployment(hcp.Namespace)
+	if _, err := createOrUpdate(ctx, r, deployment, func() error {
+		return snapshotcontroller.ReconcileOperatorDeployment(deployment, params)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile CSI snapshot controller operator deployment: %w", err)
+	}
+
+	sa := manifests.CSISnapshotControllerOperatorServiceAccount(hcp.Namespace)
+	if _, err := createOrUpdate(ctx, r, sa, func() error {
+		return snapshotcontroller.ReconcileOperatorServiceAccount(sa, params)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile CSI snapshot controller operator service account: %w", err)
+	}
+
+	role := manifests.CSISnapshotControllerOperatorRole(hcp.Namespace)
+	if _, err := createOrUpdate(ctx, r, role, func() error {
+		return snapshotcontroller.ReconcileOperatorRole(role, params)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile CSI snapshot controller operator role: %w", err)
+	}
+
+	roleBinding := manifests.CSISnapshotControllerOperatorRoleBinding(hcp.Namespace)
+	if _, err := createOrUpdate(ctx, r, roleBinding, func() error {
+		return snapshotcontroller.ReconcileOperatorRoleBinding(roleBinding, params)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile CSI snapshot controller operator roleBinding: %w", err)
+	}
+
+	// TODO: create custom kubeconfig to the guest cluster + RBAC
```

Comment:
Is this TODO still valid btw?

Reply:
Yes! snapshot-controller should eventually have its own kubeconfig and have permissions to do only "snapshot stuff" in the guest cluster.